components/relevancy/src/lib.rs

```diff
@@ -47,9 +57,53 @@ impl RelevancyStore {
     ///
     ///  This method may execute for a long time and should only be called from a worker thread.
     #[handle_error(Error)]
-    pub fn ingest(&self, _top_urls_by_frecency: Vec<String>) -> ApiResult<()> {
+    pub fn ingest(&self, top_urls_by_frecency: Vec<String>) -> ApiResult<HashMap<Interest, u32>> {
         populate_interests::ensure_interest_data_populated(&self.db)?;
-        todo!()
+        let mut interest_counter: HashMap<Interest, u32> = HashMap::new();
+
+        let hashed_top_urls: Vec<UrlHash> = top_urls_by_frecency
+            .into_iter()
+            .filter_map(|url| hash_url(&url))
+            .collect();
+
+        let classified_urls = RelevancyStore::get_url_interest_data();
+        for classified_url in classified_urls {
+            if let Some(url) = &classified_url.url {
+                match hashed_top_urls.contains(url) {
+                    true => {
+                        *interest_counter.entry(classified_url.interest).or_insert(0) += 1;
+                    }
+                    false => {
+                        *interest_counter.entry(Interest::Inconclusive).or_insert(0) += 1;
+                    }
+                }
+            } else {
+                *interest_counter.entry(Interest::Inconclusive).or_insert(0) += 1;
+            }
+        }
```

Comment:
I tried to use Inconclusive in the case we don't have the url classified or we can't hash it, but it looks goofy...

Reply:
Yeah, maybe we should just  skip the URL in this case.